var/spack/repos/builtin/packages/geopm/package.py

```diff
@@ -0,0 +1,132 @@
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Geopm(Package):
+
+    """GEOPM is an extensible power management framework targeting HPC.
+    The GEOPM package provides libgeopm, libgeopmpolicy and applications
+    geopmctl and geopmpolicy. GEOPM is designed to be extended for new control
+    algorithms and new hardware power management features via its plugin
+    infrastructure.
+
+    Note: GEOPM interfaces with hardware using Model Specific Registers (MSRs).
+    For propper usage make sure MSRs are made available directly or via the
+    msr-safe kernel module by your administrator."""
+
+    homepage = "https://geopm.github.io"
+    url      = "https://github.com/geopm/geopm/releases/download/v0.4.0/geopm-0.4.0.tar.gz"
+
+    # Add additional proper versions and checksums here. "spack checksum geopm"
+    version('0.4.0', 'd4cc8fffe521296dab379857d7e2064d')
+    version('0.3.0', '568fd37234396fff134f8d57b60f2b83')
+    version('master', git='https://github.com/geopm/geopm.git', branch='master')
+    version('develop', git='https://github.com/geopm/geopm.git', branch='dev')
+
+    # Variants reflecting most ./configure --help options
+    variant('debug', default=False, description='Enable debug.')
+    variant('coverage', default=False, description='Enable for test coverage support and enable debug.')
+    variant('procfs', default=True, description='Use procfs depending on operating systems.')
+    variant('mpi', default=True, description='To disable dependent MPI components.')
+    variant('fortran', default=True, description='Build fortran interface.')
+    variant('doc', default=True, description='Create man pages with ronn.')
+    variant('openmp', default=True, description='Build with OpenMP.')
+    variant('ompt', default=False, description='Use OpenMP Tool interface.')
+    variant('hwloc', default=True, description='Build with hwloc.')
+    variant('gnu-ld', default=False, description='Assume C uses gnu-ld.')
```

Comment:
does this variant change the resulting installed package or is it for build only?

Reply:
This is for the build process. How to properly change this if this is the case?

Reply:
> This is for the build process. How to properly change this if this is the case?

@alalazo @adamjstewart @tgamblin @scheibelp any ideas? Can we get something from spec/compilers?

p.s. Personally, i think we should avoid variants which do not change the state of the installed package (features/debug vs release/etc).

Reply:
I agree, but I don't know any other way to do this at the moment.